README.md

```diff
@@ -1,17 +1,227 @@
+
+### Create a `prediction` endpoint
+
+**📝 Add a `/predict` endpoint to `api/app.py`.**
+
+👉 Remember, throughout this challenge, you are able to test your endpoint easily at http://localhost:8000/docs
+
+### Load the pipeline
+
+**📝 In the prediction endpoint, load the trained pipeline that is provided.**
+
+👉 You can use the following code:
+
+``` python
+import joblib
+
+pipeline = joblib.load("model.joblib")
+```
+
+### API definition
+
+**📝 Here is the definition of the API that we want to create.**
+
+The API will take as input the following parameters:
+- acousticness: float
+- danceability: float
+- duration_ms: int
+- energy: float
+- explicit: int
+- id: string
+- instrumentalness: float
+- key: int
+- liveness: float
+- loudness: float
+- mode: int
+- name: string
+- release_date: string
+- speechiness: float
+- tempo: float
+- valence: float
+- artist: string
+
+💡 The pipeline contained in the `model.joblib` file expects for its predictions a `DataFrame` with the following data types:
+
+```
+acousticness        float64
+danceability        float64
+duration_ms           int64
+energy              float64
+explicit              int64
+id                   object
+instrumentalness    float64
+key                   int64
+liveness            float64
+loudness            float64
+mode                  int64
+name                 object
+release_date         object
+speechiness         float64
+tempo               float64
+valence             float64
+artist               object
+```
+
+👉 The pipeline expects these parameters to be provided in this exact same order 🚨
+
+The API will return a json dictionary with a `pred` key containing the prediction of the popularity of the track as a `float`
+
+``` json
+{
+  "pred": 22.245404331801364
+}
+```
+
+### Handle the API parameters
+
+**📝 Add to your `/predict` endpoint the parameters that are required in order to make a prediction. Then create a `DataFrame` variable named `X_pred` from these parameters.**
+
+👉 Remember that `FastAPI` only provides `string` variables as parameters to the endpoint
+
+👉 Remember that the parameters need to be built into the `DataFrame` in the exact **same order** as the one required by the pipeline 🚨
+
+💡 One way of building a `DataFrame` from individual values is to create it from a dictionary of lists
+
+👉 Once `X_pred` is built, you may `print` it and have a look at the output of the `uvicorn` command in order to make sure that it is correctly created
+
+### Make a prediction
+
+**📝 Now that you have a trained pipeline and a `DataFrame` containing one observation, let's make a prediction and return it.**
+
+👉 Remember that `FastAPI` can only handle simple data types as returned `json` values
+
+Test your API using http://localhost:8000/docs and make sure that it works correctly before proceeding to put it in production.
+
+## API in production
+
+**📝 Push your API to production on the hosting service of your choice.**
```

Comment:
OK but the Makefile is for GCR only!!

Reply:
I added the instructions for GCP in a collapsed paragraph 🙏